src/fn_call.rs

```diff
@@ -214,11 +216,35 @@ pub trait EvalContextExt<'a, 'mir, 'tcx: 'a + 'mir>: crate::MiriEvalContextExt<'
                 //
                 // `libc::syscall(NR_GETRANDOM, buf.as_mut_ptr(), buf.len(), GRND_NONBLOCK)`
                 // is called if a `HashMap` is created the regular way.
-                match this.read_scalar(args[0])?.to_usize(this)? {
-                    318 | 511 => {
-                        return err!(Unimplemented(
-                            "miri does not support random number generators".to_owned(),
-                        ))
+                match this.read_scalar(args[0])?.to_usize(this)? as i64 {
+                    // SYS_getrandom on x86_64 and x86 respectively
+                    318 | 355 => {
+                        match this.machine.rng.as_ref() {
+                            Some(rng) => {
+                                let ptr = this.read_scalar(args[1])?.to_ptr()?;
+                                let len = this.read_scalar(args[2])?.to_usize(this)?;
+
+                                // The only supported flags are GRND_RANDOM and GRND_NONBLOCK,
+                                // neither of which have any effect on our current PRNG
+                                let _flags = this.read_scalar(args[3])?.to_i32()?;
+
+                                let mut data = vec![0; len as usize];
+                                rng.borrow_mut().fill_bytes(&mut data);
+
+                                this.memory_mut().get_mut(ptr.alloc_id)?
+                                    .write_bytes(tcx, ptr, &data)?;
```

Comment:
Unfortunately, I think that would require extensive refactoring to other parts of Miri. Calling `fill_bytes` requires that we borrow the `rng` field of `Evaluator`, while calling `get_mut` takes a mutable reference. I don't see a way of getting this compile without changing `memory_mut` to take `&self` instead of `&mut self` (e.g. by using a `RefCell` internallu)

Reply:
Oh right, we don't have access to the `memory` field